website/docs/installation/reverse-proxy.md

```diff
@@ -2,40 +2,75 @@
 title: Reverse-proxy
 ---
 
-If you want to access authentik behind a reverse-proxy, use a config like this. It is important that Websocket is enabled, so that Outposts can connect.
+:::info
+Since authentik uses WebSockets to communicate with Outposts, it does not support HTTP/1.0 reverse-proxies. The HTTP/1.0 specification does not officially support WebSockets or protocol upgrades, though some clients may allow it.
+:::
+
+If you want to access authentik behind a reverse-proxy, there are a few headers that must be passed upstream:
+- `X-Forwarded-Proto`: Tells authentik and Proxy Providers if they are being served over a HTTPS connection.
+- `X-Forwarded-For`: Without this, authentik will not know the IP addresses of clients.
+- `Host`: Required for various security checks, WebSocket handshake, and Outpost and Proxy Provider communication.
+- `Connection: Upgrade` and `Upgrade: WebSocket`: Required for requests to the `/ws/client/` endpoint under HTTP/1.1, to upgrade to the WebSocket protcol.
+
+The following nginx configuration can be used as a starting point for your own configuration.
 
 ```
+# Upstream where your authentik server is hosted.
+upstream authentik {
+    server <hostname of your authentik server>:9443;
+    # Improve performance by keeping some connections alive.
+    # https://nginx.org/r/keepalive
+    keepalive 10;
+}
+
+# Provided by default in most Linux package distributions.
 map $http_upgrade $connection_upgrade {
     default upgrade;
     ''      close;
 }
 
 server {
-    # Server config
+    # HTTP server config
     listen 80;
     server_name sso.domain.tld;
 
-    # 301 to SSL
+    # 301 redirect to HTTPS
     location / {
             return 301 https://$host$request_uri;
     }
 }
 server {
-    # Server config
+    # HTTPS server config
     listen 443 ssl http2;
     server_name sso.domain.tld;
 
-    # SSL Certs
+    # TLS certificates
     ssl_certificate /etc/letsencrypt/live/domain.tld/fullchain.pem;
     ssl_certificate_key /etc/letsencrypt/live/domain.tld/privkey.pem;
 
     # Proxy site
     location / {
-        proxy_pass https://<hostname of your authentik server>:9443;
+        proxy_pass https://authentik;
         proxy_http_version 1.1;
-        proxy_set_header X-Forwarded-Proto https;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header Host $host;
+        # Use HTTP/1.1 keepalive by unsetting the Connection header.
+        # It is set to "close" by default.
+        # https://nginx.org/r/proxy_set_header
+        proxy_set_header Connection "";
+    }
+
+    # WebSocket
+    location /ws/client/ {
```

Comment:
This should be `/ws/` since there's another websocket endpoint besides `/ws/client/`, namely `/ws/outpost/<uuid>/` for Outposts (also I don't get the point of having a separate location block for websockets compared to just setting the headers correctly in the main block)

Reply:
I could only see `client` when glancing at the code/page requests, thanks.
I don't know why I didn't think of it before, but you could just rewrite the mapping to replace `close` with the empty string, and both blocks could be combined. This is just how I have been doing it in all my configs, since I never declared my own mapping and used the mapping from `nginx.conf` provided by the package maintainer. I will change it.
My previous train of thought was that on most configurations and with the current setup, the `$connection_upgrade` mapping is `close` for all other requests than ones with the `Upgrade` header set, so merging them into one would mean that those requests would not be persistent.